### PR TITLE
Provide Valid Lambda Function Memory Values for use in Parameters

### PR DESCRIPTION
**Pull request by ink-adavison**
This is a retry of PR435 after I thoroughly messed up the commit history there. This PR should be much cleaner.

Lambda functions can only have certain amounts of memory in multiples of 64
http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-memorysize

This commit allows MEMORY_VALUES to be used against Parameters' AllowedValues attribute, so that the Cloudformation web console form has a drop down selection of memory values in the same way it currently is in the Lambda web console.

I've updated the Lambda example to demonstrate.

I'm aware this could become unwieldy if AWS increase the amounts of memory available to Lambda functions, but if that occurs, this could be reverted or compared with any UI changes in the Lambda web console.

This also includes the updated validation as requested by phobologic

### CODE REVIEW

File: troposphere/awslambda.py [hunk 1]
```diff
@@ -1,6 +1,23 @@
 from . import AWSObject, AWSProperty
 from .validators import positive_integer
 
+MEMORY_VALUES = [128, 192, 256, 320, 384, 448, 512,
```
Comment (markpeek) [inline troposphere/awslambda.py]: Could you shorten this to:
```MEMORY_VALUES = [x for x in range(128, 1600, 64)]```

**Comment by ink-adavison** [comment]
Thanks for suggesting that, it is much tidier, yes.

**Comment by markpeek** [comment]
Thanks!
